lib/js/jsbigints.nim

```diff
@@ -0,0 +1,130 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func newBigInt*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".}
+  ## Constructor for `JsBigInt`.
+
+func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".}
+  ## Constructor for `JsBigInt`.
+
+func toLocaleString*(this: JsBigInt): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: cstring): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): cstring {.importjs: "#.toString()".}
```

Comment:
```nim
func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString

func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".}
  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString

func `$`*(this: JsBigInt): string = $toString(this)
```


otherwise echo + friends behaves badly

after this change:
```nim
when true:
  import std/jsbigints
  proc main()=
    let a = "10".cstring.newBigInt
    echo $a
    echo a
  main()
```
prints:
```
10
10
```

before this change:
prints:
```
10

```